src/main/java/org/example/ContinuousIntegrationServer.java

```diff
@@ -63,4 +71,30 @@ public Object handle(Request request, Response response) {
             return "CI job done";
         }
     }
+
+    public static void runBuild(GitHandler gitHandler) {
```

Comment:
In my part you will have the correct branch already checked out and pulled, so you don't need to worry about git commands in here
Especially since you need to build the branch to which the changes were just made not always main

Reply:
ah ok sounds good! will fix this now!